lib/base_service.js

```diff
@@ -145,13 +163,16 @@ BaseService.prototype._getOptions = function(opts) {
     args.build = args._.indexOf('build') !== -1 || args.deployRepo;
     args.dockerStart = args._.indexOf('docker-start') !== -1;
     args.dockerTest = args._.indexOf('docker-test') !== -1;
-    args.deployRepo = args.deployRepo || args.build;
+    args.generate = args._.indexOf('generate') !== -1;
+    args.deployRepo = args.deployRepo || args.build || args.building;
 
-    if (args.build && args.dockerStart || args.build && args.dockerTest
-    || args.dockerStart && args.dockerTest) {
+    /*jshint bitwise: false*/
+    if (!(args.build ^ args.dockerStart ^ args.dockerTest ^ args.generate) &&
```

Comment:
@gwicke Eh, we still haven't deprecated node 0.1x for service-runner so that's not going to fly, but I agree it might be slightly obvious (even though in both cases one needs to think what's going on). Luckily the next line explains it pretty clearly :)